docs/pages/kubernetes-access/discovery/google-cloud.mdx

```diff
@@ -0,0 +1,464 @@
+---
+title: Teleport GKE Auto-Discovery (Preview)
+description: How to configure auto-discovery of Google Kubernetes Engine clusters in Teleport.
+---
+
+The Teleport Discovery Service can automatically register your Google Kubernetes
+Engine (GKE) clusters with Teleport. With Teleport Kubernetes Auto-Discovery,
+you can configure the Teleport Kubernetes Service and Discovery Service once,
+then create GKE clusters without needing to register them with Teleport after
+each creation.  
+
+In this guide, we will show you how to get started with Teleport Kubernetes
+Auto-Discovery for GKE.
+
+## Overview
+
+(!docs/pages/kubernetes-access/discovery/includes/step-description.mdx!)
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- A Google Cloud account with permissions to create GKE clusters, IAM roles, and
+  service accounts.
+- The `gcloud` CLI tool. Follow the [Google Cloud documentation
+  page](https://cloud.google.com/sdk/docs/install-sdk) to install and
+  authenticate to `gcloud`.
+- One or more GKE clusters running. Your Kubernetes user must have permissions
+  to create `ClusterRole` and `ClusterRoleBinding` resources in your clusters.
+- A Linux host where you will run the Teleport Discovery and Kubernetes
+  services. You can run this host on any cloud provider or even use a local
+  machine.
+
+(!docs/pages/includes/tctl.mdx!)
+
+## Step 1/3. Obtain Google Cloud credentials
+
+The Teleport Discovery Service and Kubernetes Service use a Google Cloud service
+account to discover GKE clusters and manage access from Teleport users. In this
+step, you will create a service account and download a credentials file for the
+Teleport Discovery Service.
+
+### Create an IAM role for the Discovery Service
+
+The Teleport Discovery Service needs permissions to retrieve GKE clusters
+associated with your Google Cloud project.
+
+To grant these permissions, create a file called `GKEKubernetesAutoDisc.yaml`
+with the following content:
+
+```yaml
+title: GKE Cluster Discoverer
+description: "Get and list GKE clusters"
+stage: GA
+includedPermissions:
+- container.clusters.get
+- container.clusters.list
+```
+
+Create the role, replacing `project-id` with the name of your project:
+
+```code
+$ gcloud iam roles create GKEKubernetesAutoDisc \
+  --project=project-id \
+  --file=GKEKubernetesAutoDisc.yaml
+```
+
+### Create an IAM role for the Kubernetes Service
+
+The Teleport Kubernetes Service needs Google Cloud IAM permissions in order to
+forward user traffic to your GKE clusters. 
+
+Create a file called `GKEAccessManager.yaml` with the following content:
+
+```yaml
+title: GKE Cluster Access Manager
+description: "Manage access to GKE clusters"
+stage: GA
+includedPermissions:
```

Comment:
Kubernetes Service also needs `container.clusters.get`.

The AKS and EKS guides show this step as unique permissions for both services. Keeping everything uniform we should show the permissions mixed here or separated in the other two cases to maintain consistency.